### PR TITLE
Adds possibility to customize crates

### PR DESCRIPTION
**Pull request by fira**
# About the pull request

You can now customize non-secure crates using packaging wrap, by tagging them and changing their appearance to a list of common crates. The tag works as a labeler label, and is also displayed in the overwatch supply drop alert.

# Explain why it's good for the game
If anything, it's that these Squad crates are just sitting there unused. I don't know if this feature will be used much by live Reqs.

# Changelog
:cl:
add: You can now use package wrap to change standard crates visuals.
add: You can now label crate with package wrap, which will also be shown during overwatch supply drop alert.
mapadd: Added more packaging wrap to Requisitions.
/:cl:

### CODE REVIEW

File: code/datums/components/crate_tag.dm [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		crate.icon_closed = initial(masquarade_type.icon_closed)
+		if(crate.opened)
+			crate.icon_state = crate.icon_opened
+		else
+			crate.icon_state = crate.icon_closed
+	if(name)
+		parent.AddComponent(/datum/component/label, name)
+		src.name = name // Keep it around additionally for notifications
+
+/datum/component/crate_tag/proc/notify_squad(datum/source, datum/squad/squad)
```
Comment (Zonespace27) [inline code/datums/components/crate_tag.dm:32]: ///

File: code/datums/components/crate_tag.dm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	var/name
+
+/datum/component/crate_tag/Initialize(name, obj/structure/closet/crate/masquarade_type)
+	var/obj/structure/closet/crate/crate = parent
+	if(!istype(crate))
+		return COMPONENT_INCOMPATIBLE
+	setup(name, masquarade_type)
+	RegisterSignal(parent, COMSIG_STRUCTURE_CRATE_SQUAD_LAUNCHED, PROC_REF(notify_squad))
+
+/datum/component/crate_tag/InheritComponent(datum/component/C, i_am_original, name, obj/structure/closet/crate/masquarade_type)
```
Comment (Zonespace27) [inline code/datums/components/crate_tag.dm:13]: one letter vars

File: code/modules/cm_tech/droppod/supply.dm [hunk 3]
```diff
@@ -1,16 +1,31 @@
 /obj/structure/droppod/supply
-	name = "\improper USCM requisitions droppod"
+	name = "\improper USCM requisitions package"
 	drop_time = 10 SECONDS
 	dropping_time = 2 SECONDS
 	open_time = 2 SECONDS
+	var/obj/structure/package
```
Comment (Zonespace27) [inline code/modules/cm_tech/droppod/supply.dm:6]: code doc

File: code/modules/cm_tech/droppod/supply.dm [hunk 4]
```diff
@@ -1,16 +1,31 @@
...
+/obj/structure/droppod/supply/Initialize(mapload, obj/structure/closet/crate/package)
+	. = ..()
+	if(!istype(package))
+		return INITIALIZE_HINT_QDEL
+	package.forceMove(src)
+	src.package = package
+
+/obj/structure/droppod/supply/Destroy()
+	. = ..()
+	package = null
```
Comment (Zonespace27) [inline code/modules/cm_tech/droppod/supply.dm:17]:
```suggestion
	package = null
	return ..()
```

File: code/modules/recycling/sortingmachinery.dm [hunk 5]
```diff
@@ -267,18 +267,34 @@
...
+			target.AddComponent(/datum/component/crate_tag, label, crate.crate_customizing_types[chosen_type])
 			src.amount -= 3
-			user.visible_message("\The [user] wraps \a [target] with \a [src].",\
-			SPAN_NOTICE("You wrap \the [target], leaving [amount] units of paper on \the [src]."),\
-			"You hear someone taping paper around a large object.")
-		else if(src.amount < 3)
-			to_chat(user, SPAN_WARNING("You need more paper."))
+		else
+			if (src.amount > 3 && !crate.opened)
+				var/obj/structure/bigDelivery/P = new /obj/structure/bigDelivery(get_turf(crate.loc))
```
Comment (Zonespace27) [inline code/modules/recycling/sortingmachinery.dm:288]: one letter

File: code/modules/recycling/sortingmachinery.dm [hunk 6]
```diff
@@ -267,18 +267,34 @@
...
+				return
+			target.AddComponent(/datum/component/crate_tag, label, crate.crate_customizing_types[chosen_type])
 			src.amount -= 3
-			user.visible_message("\The [user] wraps \a [target] with \a [src].",\
-			SPAN_NOTICE("You wrap \the [target], leaving [amount] units of paper on \the [src]."),\
-			"You hear someone taping paper around a large object.")
-		else if(src.amount < 3)
-			to_chat(user, SPAN_WARNING("You need more paper."))
+		else
+			if (src.amount > 3 && !crate.opened)
```
Comment (Zonespace27) [inline code/modules/recycling/sortingmachinery.dm:287]: remove src

File: code/modules/recycling/sortingmachinery.dm [hunk 7]
```diff
@@ -267,18 +267,34 @@
...
+				to_chat(user, SPAN_WARNING("You cannot customize this kind of crate."))
+				return
+			var/label = tgui_input_text(user, "Give the crate a new logistic tag:", "Customizing")
+			if(!label || !user.Adjacent(target) || !target.z)
+				return
+			var/chosen_type = tgui_input_list(user, "Select the kind of crate to make this into:", "Customizing", crate.crate_customizing_types)
+			if(!chosen_type || !ispath(crate.crate_customizing_types[chosen_type]) || !user.Adjacent(target) || !target.z)
+				return
+			target.AddComponent(/datum/component/crate_tag, label, crate.crate_customizing_types[chosen_type])
 			src.amount -= 3
```
Comment (Zonespace27) [inline code/modules/recycling/sortingmachinery.dm:285]: ditto

File: code/modules/recycling/sortingmachinery.dm [hunk 8]
```diff
@@ -267,18 +267,34 @@
...
-			"You hear someone taping paper around a large object.")
-		else if(src.amount < 3)
-			to_chat(user, SPAN_WARNING("You need more paper."))
+		else
+			if (src.amount > 3 && !crate.opened)
+				var/obj/structure/bigDelivery/P = new /obj/structure/bigDelivery(get_turf(crate.loc))
+				P.icon_state = "deliverycrate"
+				P.wrapped = crate
+				crate.forceMove(P)
+				src.amount -= 3
```
Comment (Zonespace27) [inline code/modules/recycling/sortingmachinery.dm:292]: ditto

File: code/modules/recycling/sortingmachinery.dm [hunk 9]
```diff
@@ -267,18 +267,34 @@
...
+				user.visible_message("\The [user] wraps \a [target] with \a [src].",\
+				SPAN_NOTICE("You wrap \the [target], leaving [amount] units of paper on \the [src]."),\
```
Comment (Zonespace27) [inline code/modules/recycling/sortingmachinery.dm:294]: \the and \a are unneeded

File: code/modules/recycling/sortingmachinery.dm [hunk 10]
```diff
@@ -267,18 +267,34 @@
...
+				"You hear someone taping paper around a large object.")
+			else if(src.amount < 3)
```
Comment (Zonespace27) [inline code/modules/recycling/sortingmachinery.dm:296]: src

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/datums/components/crate_tag.dm (hunk 2 above)
Comment (fira) [reply code/datums/components/crate_tag.dm:13]: It's parent signature

**Review by realforest2001 — APPROVED** [review]

**Review by Zonespace27 — APPROVED** [review]

**Pull request merged**